### PR TITLE
Code Insights: [FE] Create live preview for capture creation UI

### PR DESCRIPTION
**Pull request by vovakulikov**
Follow up for https://github.com/sourcegraph/sourcegraph/pull/28654

**Series of PR** 
1. [This one creates main layouts for capture group creation UI](https://github.com/sourcegraph/sourcegraph/pull/28654)
2. This one
3. [Improve live preview chart and connect it to gql api](https://github.com/sourcegraph/sourcegraph/pull/28757)

This PR updates the live preview component for all creation UI pages in a way to support content with dynamic height. Besides this, it adds a creation UI live preview chart for capture group insight and API for fetching live preview data. 

Note that this PR updates just visual components. Capture group insight live preview chart will be properly connected to API in a separate PR 

## Todo
- [x] Create a live preview chart
- [x] Fix chart styles for different screen sizes 
- [x] Adjust appearance logic

### CODE REVIEW

File: client/web/src/enterprise/insights/components/form/hooks/utils/use-async-validation.ts [hunk 1]
```diff
@@ -84,7 +84,7 @@ export function useAsyncValidation<FieldValue>(
     const asyncValidationPipeline = useCallback(
         (validationEvents: Observable<AsyncValidationEvent<FieldValue>>) =>
             validationEvents.pipe(
-                debounceTime(ASYNC_VALIDATION_DEBOUNCE_TIME),
+                // debounceTime(ASYNC_VALIDATION_DEBOUNCE_TIME),
```
Comment (unclejustin) [inline client/web/src/enterprise/insights/components/form/hooks/utils/use-async-validation.ts:87]: Probably want to delete this line.

File: client/web/src/enterprise/insights/core/backend/gql-api/gql-handlers/get-capture-group-insight-preivew.ts [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                    query: input.query,
+                    label: '',
+                    repositoryScope: { repositories: input.repositories },
+                    generatedFromCaptureGroups: true,
+                    timeScope: { stepInterval: { unit, value } },
+                },
+            },
+        })
+        .then(({ data, error }) => {
+            console.log(data)
```
Comment (unclejustin) [inline client/web/src/enterprise/insights/core/backend/gql-api/gql-handlers/get-capture-group-insight-preivew.ts:44]: I assume this is just a placeholder until we wire it up for real?

File: client/web/src/enterprise/insights/pages/insights/creation/search-insight/components/search-insight-creation-content/SearchInsightCreationContent.tsx [hunk 3]
```diff
@@ -72,13 +71,11 @@ export const SearchInsightCreationContent: React.FunctionComponent<SearchInsight
...
     }
 
-    const repositoriesList = getSanitizedRepositories(repositories.input.value)
-
     // If some fields that needed to run live preview  are invalid
     // we should disabled live chart preview
     const allFieldsForPreviewAreValid =
         repositories.meta.validState === 'VALID' &&
-        repositoriesList.length > 0 &&
+        // repositoriesList.length > 0 &&
```
Comment (unclejustin) [inline client/web/src/enterprise/insights/pages/insights/creation/search-insight/components/search-insight-creation-content/SearchInsightCreationContent.tsx:78]: Should we delete this?

**Review by unclejustin — APPROVED** [review]
@vovakulikov  Just a handful of code cleanup questions.

File: client/web/src/enterprise/insights/core/backend/gql-api/gql-handlers/get-capture-group-insight-preivew.ts (hunk 2 above)
Comment (vovakulikov) [reply client/web/src/enterprise/insights/core/backend/gql-api/gql-handlers/get-capture-group-insight-preivew.ts:44]: Yeah. This is just a placeholder here. This part is implemented here https://github.com/sourcegraph/sourcegraph/pull/28757

File: client/web/src/enterprise/insights/pages/insights/creation/search-insight/components/search-insight-creation-content/SearchInsightCreationContent.tsx (hunk 3 above)
Comment (vovakulikov) [reply client/web/src/enterprise/insights/pages/insights/creation/search-insight/components/search-insight-creation-content/SearchInsightCreationContent.tsx:78]: Good catch. Thanks.

**Pull request merged**